middleware/http/opa/opa_middleware.go

```diff
@@ -0,0 +1,178 @@
+package opa
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/dapr/components-contrib/middleware"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/valyala/fasthttp"
+)
+
+type opaMiddlewareMetadata struct {
+	Rego            string `json:"rego"`
+	IncludedHeaders string `json:"includedHeaders"`
+}
+
+// NewOpaMiddleware returns a new Open Policy Agent middleware
+func NewOpaMiddleware(logger logger.Logger) *Middleware {
```

Comment:
Change to `NewMiddleWare`? That would remove the stutter in downstream call (i.e. `opa.NewMiddleWare()`)